addons/account_followup/account_followup.py

```diff
@@ -282,39 +246,39 @@ def get_followup_table_html(self, cr, uid, ids, context=None):
-                    self.pool.get("mail.thread").message_post(cr, uid, 0, 
+                    responsible_partner_id = self.pool.get("res.users").browse(vals['payment_responsible_id']).partner_id.id
+                    self.env['mail.thread'].message_post(0,
                                       body = _("You became responsible to do the next action for the payment follow-up of") + " <b><a href='#id=" + str(part.id) + "&view_type=form&model=res.partner'> " + part.name + " </a></b>",
                                       type = 'comment',
-                                      subtype = "mail.mt_comment", context = context,
-                                      model = 'res.partner', res_id = part.id, 
+                                      subtype = "mail.mt_comment",
+                                      model = 'res.partner', res_id = part.id,
                                       partner_ids = [responsible_partner_id])
-        return super(res_partner, self).write(cr, uid, ids, vals, context=context)
+        return super(ResPartner, self).write(vals)
 
-    def action_done(self, cr, uid, ids, context=None):
-        return self.write(cr, uid, ids, {'payment_next_action_date': False, 'payment_next_action':'', 'payment_responsible_id': False}, context=context)
+    @api.model
+    def action_done(self):
+        return self.write({'payment_next_action_date': False, 'payment_next_action': '', 'payment_responsible_id': False})
 
-    def do_button_print(self, cr, uid, ids, context=None):
-        assert(len(ids) == 1)
-        company_id = self.pool.get('res.users').browse(cr, uid, uid, context=context).company_id.id
+    def do_button_print(self):
+        assert(len(self.ids) == 1)
+        company_id = self.env.users.company_id.id
         #search if the partner has accounting entries to print. If not, it may not be present in the
         #psql view the report is based on, so we need to stop the user here.
-        if not self.pool.get('account.move.line').search(cr, uid, [
-                                                                   ('partner_id', '=', ids[0]),
-                                                                   ('account_id.internal_type', '=', 'receivable'),
-                                                                   ('reconciled', '=', False),
-                                                                   ('company_id', '=', company_id),
-                                                                  ], context=context):
+        if not self.env['account.move.line'].search([('partner_id', '=', self.id),
+                                                   ('account_id.internal_type', '=', 'receivable'),
+                                                   ('reconciled', '=', False),
+                                                   ('company_id', '=', company_id), ]):
```

Comment:
Maybe add limit=1 to avoid bringing in all of the record, and use search_count so it actually brings no data, just 0 or 1 ?